app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTasksFragment.kt

```diff
@@ -218,54 +245,59 @@ class SuggestedEditsTasksFragment : Fragment() {
+            binding.editStreakStatsView.setTitle(resources.getString(R.string.suggested_edits_edit_streak_label_text))
+            binding.editStreakStatsView.setDescription(resources.getQuantityString(R.plurals.suggested_edits_edit_streak_detail_text,
                 viewModel.latestEditStreak, viewModel.latestEditStreak))
-            binding.editStreakStatsView.setDescription(resources.getString(R.string.suggested_edits_edit_streak_label_text))
         }
 
         if (viewModel.totalContributions == 0) {
-            binding.userStatsClickTarget.isEnabled = false
-            binding.userStatsViewsGroup.visibility = GONE
-            binding.onboardingImageView.visibility = VISIBLE
-            binding.onboardingTextView.visibility = VISIBLE
-            binding.onboardingTextView.text = StringUtil.fromHtml(getString(R.string.suggested_edits_onboarding_message, AccountUtil.userName))
+            binding.contributionsContainer.isVisible = false
+            binding.statsDivider.isVisible = false
+            binding.messageCard.isVisible = true
+            binding.messageCard.setOnboarding(getString(R.string.suggested_edits_onboarding_message, AccountUtil.userName))
         } else {
-            binding.userStatsViewsGroup.visibility = VISIBLE
-            binding.onboardingImageView.visibility = GONE
-            binding.onboardingTextView.visibility = GONE
-            binding.userStatsClickTarget.isEnabled = true
-            binding.userNameView.text = AccountUtil.userName
-            binding.contributionsStatsView.setTitle(viewModel.totalContributions.toString())
-            binding.contributionsStatsView.setDescription(resources.getQuantityString(R.plurals.suggested_edits_contribution, viewModel.totalContributions))
-            if (Prefs.showOneTimeSequentialUserStatsTooltip) {
-                showOneTimeSequentialUserStatsTooltips()
-            }
+            binding.contributionsContainer.isVisible = true
+            binding.statsDivider.isVisible = true
+            val contributionsStatsViewPluralRes = if (ContributionsDashboardHelper.contributionsDashboardEnabled)
+                R.plurals.suggested_edits_edit_frequency else R.plurals.suggested_edits_contribution
+            binding.editsCountStatsView.setTitle(resources.getQuantityString(contributionsStatsViewPluralRes, viewModel.totalContributions))
+            binding.editsCountStatsView.setDescription(viewModel.totalContributions.toString())
+            // TODO: add the sequential tooltips back after the experiment code is removed.
+//            if (Prefs.showOneTimeSequentialUserStatsTooltip) {
+//                 showOneTimeSequentialUserStatsTooltips()
+//            }
+        }
+
+        binding.donorHistoryContainer.isVisible = true
+        if (!ContributionsDashboardHelper.contributionsDashboardEnabled) {
+            binding.donorHistoryContainer.isVisible = false
+            binding.statsDivider.isVisible = false
         }
```

Comment:
I think we should hide `donorHistoryContainer ` by default and only display it when `ContributionsDashboardHelper.contributionsDashboardEnabled` is true. Currently we are only checking `ContributionsDashboardHelper.contributionsDashboardEnabled` when `viewModel.totalContributions != 0`. This means that if the user has not made any contributions we aren't validating this flag, which allows any regions to view the donor history container even without contributions.

Reply:
Thanks! It makes sense to me. Fixed.